File: engine/runner.py
import subprocess
import time

# Number of runs
total_runs = 6

for i in range(1, total_runs + 1):
    print(f"\n🔄 Running iteration {i}...")

    try:
        # Execute main.js using Node.js
        result = subprocess.run(
            ["node", "main.js"],
            check=True,
            capture_output=True,
            text=True
        )
        print(f"✅ Run {i} completed successfully.")
        print(result.stdout)

    except subprocess.CalledProcessError as e:
        print(f"⚠️ Run {i} failed. Skipping to next...")
        print(f"Error Output:\n{e.stderr.strip()}")

    except Exception as e:
        print(f"❌ Unexpected error in run {i}: {e}")
        continue

    # Optional: add a short delay between runs (e.g., 1 second)
    time.sleep(1)

print("\n🏁 All runs completed.")

File: Creator/runner.js
// batch_creator.js
// Usage: node batch_creator.js

const { exec } = require('child_process');

const totalRuns = 10;
let currentRun = 0;

console.log(`🚀 Starting batch creator — will run creator.js ${totalRuns} times.\n`);

function runCreator() {
  if (currentRun >= totalRuns) {
    console.log(`✅ All ${totalRuns} runs completed. Exiting.`);
    process.exit(0);
  }

  currentRun++;
  console.log(`\n⚙️  Run ${currentRun} of ${totalRuns} starting...`);

  const startTime = Date.now();
  const processInstance = exec('node creator.js', { timeout: 10 * 60 * 1000 }); // 10-min timeout safeguard

  processInstance.stdout.on('data', (data) => process.stdout.write(data));
  processInstance.stderr.on('data', (data) => process.stderr.write(data));

  processInstance.on('close', (code) => {
    const duration = ((Date.now() - startTime) / 1000).toFixed(2);

    if (code === 0) {
      console.log(`✅ Run ${currentRun} completed successfully in ${duration}s.`);
    } else {
      console.warn(`⚠️ Run ${currentRun} failed (exit code ${code}). Skipping and continuing...`);
    }

    // Small delay between runs to prevent resource overlap
    setTimeout(runCreator, 2000);
  });

  processInstance.on('error', (err) => {
    console.error(`❌ Unexpected error on run ${currentRun}: ${err.message}`);
    setTimeout(runCreator, 2000);
  });
}

// Start first run
runCreator();
